crates/aptos-rosetta/src/main.rs

```diff
@@ -13,15 +14,58 @@ use std::{
         atomic::{AtomicBool, Ordering},
         Arc,
     },
+    thread,
+    time::Duration,
 };
+use tokio::time::Instant;
+
+const REST_API_WAIT_DURATION_MS: u64 = 60;
+const TOTAL_REST_API_WAIT_DURATION_S: u64 = 300;
 
 #[tokio::main]
 async fn main() {
-    aptos_logger::Logger::new().init();
     let args: CommandArgs = CommandArgs::parse();
 
+    // If we're in online mode, we run a full node side by side
+    let _maybe_node = if let CommandArgs::Online(OnlineLocalArgs {
+        ref node_args,
+        ref online_args,
+    }) = args
+    {
+        let node_args = node_args.clone();
+        let runtime = thread::spawn(move || node_args.run());
+
+        // Wait and ensure the node is running on the URL
+        let client = aptos_rest_client::Client::new(online_args.rest_api_url.clone());
+        let mut successful = false;
+        let total_wait_duration = Duration::from_secs(TOTAL_REST_API_WAIT_DURATION_S);
+        let start = Instant::now();
+        while start.elapsed() < total_wait_duration {
+            if client.get_index().await.is_ok() {
+                successful = true;
+                break;
+            }
+
+            tokio::time::sleep(Duration::from_millis(REST_API_WAIT_DURATION_MS)).await;
+        }
+
+        // If it didn't start up, we need to crash
+        if !successful {
+            panic!(
+                "Node didn't start up on time after {} seconds",
+                TOTAL_REST_API_WAIT_DURATION_S
+            )
+        }
+
+        Some(runtime)
+    } else {
+        // Offline mode we just need to start up the logger
+        aptos_logger::Logger::new().init();
```

Comment:
But we don't want to start it in regular mode?

Reply:
This is so annoying, but the logger is global.  The node configuration should set the logger, so this only should be run in "offline" or no-node mode.